### ISSUE TITLE
Move @auth/sveltekit's dependency of @auth/core to a peerDependency

### ISSUE THREAD

**Issue by canastro** [nextauthjs/next-auth#6831]
### Description 📓

Right now @auth/sveltekit pins @auth/core dependency to whatever version is available in time of the release. This dependency should be moved to peerDependencies.

### How to reproduce ☕️

Install in your package.json the following deps:

```
    "@auth/core": "^0.5.0",
    "@auth/sveltekit": "^0.2.2",
``` 

Check the installed packages and you'll see both @auth/core at 0.5.0 and @auth/core at 0.4.0

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by balazsorban44** [comment]
Unclear why this "should" be a peer dependency. `@auth/core` is intentionally locked in to the workspace version.

We plan to maybe move provides to its own package `@auth/providers` so you wouldn't install core directly in the future though.

The main issue is #6226, so closing this in favor.

**Comment by canastro** [comment]
Alright, is just that I wanted to use something that was fixed in the core and that wasn't incompatible with the @auth/sveltekit but wasn't able due to the pinned version of the core in this dependency (had to resort to pnpm overrides). 

It feels weird that whenever the core releases a minor bug fix a release of the @auth/sveltekit will also be required. Just now the core release 0.5.0, but it happened that the @auth/sveltekit was released before, so its now pinned to 0.4.0.

Also @auth.core is a peer dependency in the [solid start package](https://github.com/nextauthjs/next-auth/blob/main/packages/frameworks-solid-start/package.json#L40-L43), I wasn't expecting different approaches between these two.
